utils/plugin/plugin_downloader.go

```diff
@@ -0,0 +1,190 @@
+package plugin
```

Comment:
nit: you are already in the `plugin` folder. File could be renamed to just `downloader.go`

Reply:
yea